src/modules/licences/controller.js

```diff
@@ -56,18 +70,48 @@ const wrapData = data => ({
   data
 });
 
+const addEarlistEndDate = licence => {
+  const dates = [
+    { key: 'expired', date: licence.licence_data_value.EXPIRY_DATE },
+    { key: 'lapsed', date: licence.licence_data_value.LAPSED_DATE },
+    { key: 'revoked', date: licence.licence_data_value.REV_DATE }
+  ];
+
+  const endedDates = dates
+    .map(date => {
+      date.date = moment(date.date, 'DD/MM/YYYY');
+      return date;
+    })
+    .filter(date => date.date.isValid())
+    .sort((l, r) => {
+      if (l.date.isSame(r.date)) return 0;
+      return l.date.isBefore(r.date) ? -1 : 1;
+    })
+    .map(date => {
+      date.date = date.date.format('YYYYMMDD');
```

Comment:
Generally the date format in the application is YYYY-MM-DD - just wondering if there's a reason for using YYYYMMDD?